builtin/providers/openstack/data_source_openstack_networking_network_v2.go

```diff
@@ -56,6 +60,7 @@ func dataSourceNetworkingNetworkV2Read(d *schema.ResourceData, meta interface{})
 	networkingClient, err := config.networkingV2Client(GetRegion(d))
 
 	listOpts := networks.ListOpts{
+		ID:       d.Get("network_id").(string),
```

Comment:
Yep, Gophercloud will ignore zero-value parameters in the GET URL. For example, here's the URL that's built for searching by name

```GET http://10.1.12.81:9696/v2.0/networks?name=tf_test_network&status=ACTIVE```

and for searching by network_id:

```GET http://10.1.12.81:9696/v2.0/networks?id=d1f4958c-dc7b-4726-8b3a-1a748b131e74&status=ACTIVE```

`tenant_id` is also unset for both, which doesn't get passed through :)